tensorly/contrib/decomposition/tt_TTOI.py

```diff
@@ -0,0 +1,173 @@
+
+import numpy as np
+from math import prod
+import tensorly as tl
+from tensorly.tt_tensor import tt_to_tensor, validate_tt_rank
+import warnings
+
+
+#Author: Lili Zheng
+
+#Reference paper: Zhou Y, Zhang AR, Zheng L, Wang Y. "Optimal high-order tensor svd via tensor-train orthogonal iteration."
+def sequential_prod_reshape(X, M_arr, direction):
+    """ Perform sequential multiplication and reshaping
+
+    Parameters
+    ----------
+    X: 2d array
+        matrix to be multiplied by a list of matrices
+        X is a (p_1p_2...p_{d-1})*p_d matrix or a p_1*(p_2...p_d) matrix
+    M_arr : list 
+        a list of 2d arrays to multiply X
+        len(M_arr)<=d-1
+        If direction == "left", M_arr{i} is a r_{i-1}p_i*r_i matrix
+        If direction == "right", M_arr{i} is a r_{d-i+1}p_{d-i+1}*r_{d-i} matrix 
+    direction : string
+        direction of the multiplication
+    Returns
+    -------
+    X_prod : 2d array
+        product matrix
+        If direction == "left" and M_arr has k elements, then X_prod is of dimension r_k*(p_{k+1}...p_d) 
+        If direction == "right" and M_arr has d-k elements, then X_prod is of dimension (p_1p_2...p_k)*r_k
+    """ 
+
+    if direction == "left":
+        X_prod = tl.matmul(tl.transpose(M_arr[0]),X)
+        for i in range(1,len(M_arr)):
+            X_prod = tl.reshape(X_prod,(tl.shape(M_arr[i])[0],-1), order = 'F')
```

Comment:
In TensorLy, we use a C ordering for unfolding (see [user-guide](http://tensorly.org/stable/user_guide/tensor_basics.html#unfolding)), I'd rather just switch to using tl.unfold() and other built-in functions for consistency.  This shouldn't matter much as long as we are consistent.

Reply:
Good point! I have changed all reshape functions to the default order (C ordering). I could not use the built-in unfolding function in tensorly since it is defined differently from the sequential unfolding we have for the tensor train structure.